transport/tchannel/options.go

```diff
@@ -33,11 +40,28 @@ var _ Option = (TransportOption)(nil)
 
 // transportOptions is suitable for conveying options to TChannel transport
 // constructors (NewTransport and NewChannelTransport).
+// At time of writing, there is only a ChannelTransport constructor, which
+// supports options like WithChannel that only apply to this constructor form.
+// The transportOptions should also be suitable, albeit with extraneous properties,
+// if used for NewTransport, which will return a Transport suitable for YARPC
+// peer lists.
+// TODO update above when NewTransport is real.
 type transportOptions struct {
-	ch     Channel
-	tracer opentracing.Tracer
-	addr   string
-	name   string
+	ch                  Channel
+	tracer              opentracing.Tracer
+	addr                string
+	name                string
+	connTimeout         time.Duration
+	connBackoffStrategy backoffapi.Strategy
+}
+
+// newTransportOptions constructs the default transport options struct
+func newTransportOptions() transportOptions {
+	return transportOptions{
+		tracer:              opentracing.GlobalTracer(),
```

Comment:
Does this have any implications if the global tracer is set later? I have generally deferred calling `opentracing.GlobalTracer()` as long as I can, ie when I actually get the tracer off the options.